server/src/domain/system-config/system-config.util.ts

```diff
@@ -0,0 +1,16 @@
+import { ServerVersion } from '../domain.constant';
+
+export function compareVersions(remoteVersionString: string, serverVersion: ServerVersion): boolean {
+  const remoteVersion = stringToVersion(remoteVersionString);
+  return remoteVersion.toString().localeCompare(serverVersion.toString(), undefined, { numeric: true }) === 1;
```

Comment:
Instead of checking `=== 1` you should do either `!!remoteVersion.toString().localeCompare(...)` or `Boolean(remoteVersion.toString().localeCompare(...))`

Reply:
I don't think it works, if the server version is bigger than the remote version `remoteVersion.toString().localeCompare(...) or Boolean(remoteVersion.toString().localeCompare(...))` returns -1 which is transforms to `true`